Implemented pong tournament mode

Taanviir opened the pull request.

Taanviir committed:
feat: implementing tournament

Taanviir committed:
feat(pong): /pong returns to home if user chooses an invalid mode

Taanviir committed:
feat: tournament can be accessed only logged in user

Taanviir committed:
feat: added /pong/tournament navigation to main.js

Taanviir committed:
feat: updated tournament design

Taanviir committed:
fix: cleaned up home.html

Taanviir committed:
feat: added tournament.js to base

Taanviir committed:
feat: added new font

- removed document.getElementById for modals and use # instead

Taanviir committed:
Merge branch 'main' into feat/tournament

Taanviir committed:
chore: removed marker from li

Taanviir committed:
Merge branch 'main' into feat/tournament

Taanviir committed:
feat: updated back_to_home design

Taanviir committed:
feat: updated tournament design and script

Taanviir committed:
Merge branch 'main' into feat/tournament

Taanviir committed:
feat: changed btn-orange to btn-accent and formatted files

Taanviir committed:
chore: updated use of fontawesome icons

Taanviir committed:
feat: updated design of tournament modal

Taanviir committed:
fix: stop game on popstate

Taanviir committed:
feat: tournament implemented

Taanviir committed:
fix: ball random speed

Taanviir committed:
chore: removed tournmanent.js error and updated design

Taanviir committed:
feat: tournament now sets difficulty

Taanviir committed:
feat: tournament now sets score to win

PaulaBalbino commented:
expection when loading tournament page 
Error when fetching the page: Cannot read properties of undefined (reading 'backdrop')

Taanviir committed:
fix: modal shows up when tournmaent is clicked

Taanviir committed:
fix: enter works for forms without spalinks

- user can use enter key to submit tournament form
- tournament form is now also responsive

Taanviir committed:
feat: tournament displays player names on either side

- refactored tournament class code

Taanviir committed:
fix: score gets reset on new game
